### PR TITLE
Added user sign in

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary

Added user sign in
## Known Issues

Can't seem to implement a happy path test without failing it, but when testing with postman it works correctly, need to review. Hence the WIP tag
## Trello Card
https://trello.com/c/yiFLog0L/3-sign-in

### CODE REVIEW

File: app/middlewares/validator.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { checkSchema, validationResult } = require('express-validator');
+
+const validate = (req, res, next) => {
+  const validationErrors = validationResult(req);
+  if (validationErrors.isEmpty()) {
+    return next();
+  }
+  const extractedErrors = [];
+  validationErrors.array().map(err => extractedErrors.push({ [err.param]: err.msg }));
```
Comment (arinaldi118) [inline app/middlewares/validator.js]: :thinking: This works but `map` is used to map elements, not to accumulate them.

Comment (tagonzalez) [reply app/middlewares/validator.js]: Changed the behavior of `map` in latest commit!

File: app/middlewares/validator.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  return res.status(422).json({
```
[the remark this answers is not in the document]
Comment (tagonzalez) [reply app/middlewares/validator.js]: Done!

File: app/services/users.js [hunk 3]
```diff
@@ -12,18 +15,47 @@ const signUp = user =>
...
+const createToken = email => {
+  const token = jwt.sign({ email }, config.common.session.secret, { expiresIn: '24h' });
+  return {
+    success: true,
+    message: 'Authentication successful',
+    token
+  };
+};
+
+const signIn = user =>
```
Comment (arinaldi118) [inline app/services/users.js]: it isn't the responsibility of the service to check if user exists or if the password is correct

File: app/services/users.js [hunk 4]
```diff
@@ -12,18 +15,47 @@ const signUp = user =>
...
+          logger.info(`New user created: ${createdUser.email}`);
+          return createdUser;
+        })
+        .catch(error => {
+          throw errors.databaseError(error.message);
+        });
+    });
+  });
+
+const createToken = email => {
```
Comment (arinaldi118) [inline app/services/users.js]: this isn't a service, this is a helper

File: app/middlewares/validator.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    return next();
+  }
+  const extractedErrors = validationErrors.array().map(err => ({ [err.param]: err.msg }));
+
+  throw errors.schemaError({ validationErrors: extractedErrors });
+};
+
+const validateSchemaAndFail = schema => [checkSchema(schema), validate];
+
+module.exports = { validate, validateSchemaAndFail };
```
Comment (arinaldi118) [inline app/middlewares/validator.js]: you dont need to export `validate`

File: app/services/users.js (hunk 4 above)
Comment (tagonzalez) [reply app/services/users.js]: Moved to utils helper!

File: app/middlewares/validator.js (hunk 5 above)
Comment (tagonzalez) [reply app/middlewares/validator.js]: Removed!

File: app/services/users.js (hunk 3 above)
Comment (tagonzalez) [reply app/services/users.js]: I am finding it difficult to find where it should go. 
Should these checks go in the middleware for sign up? Or in the controller?

Comment (tagonzalez) [reply app/services/users.js]: Logic was moved to an interactor in latest commit!

**Pull request merged**